combat-trainer.lic

```diff
@@ -1999,6 +2036,62 @@ class GameState
     @attack_overrides[weapon_skill]
   end
 
+  def set_currently_whirlwinding(setting)
+    @currently_whirlwinding = setting
+  end
+
+  def currently_whirlwinding?
+    @currently_whirlwinding
+  end
+
+  def whirlwind_trainables
+    @whirlwind_trainables
+  end
+
+  def whirlwind_offhand_name
+    weapon_training[@current_whirlwind_offhand_skill]
+  end
+
+  def last_whirlwind_offhand_weapon_name
+    weapon_training[@last_whirlwind_offhand_skill]
+  end
+
+  def update_whirlwind_weapon_info(weapon_skill)
+    @last_whirlwind_offhand_skill = @current_whirlwind_offhand_skill
+    @current_whirlwind_offhand_skill = weapon_skill
+  end
+
+  def whirlwind_offhand_skill
+    @current_whirlwind_offhand_skill
+  end
+
+  def is_whirlwind_trainable?(weapon_skill)
```

Comment:
If the input is _always_ `game_state.weapon_skill`, what do you think about using GameState's instance var instead of a function parameter?